finetune/adapter.py

```diff
@@ -127,7 +128,10 @@ def train(
 ) -> None:
     tokenizer = Tokenizer(checkpoint_dir)
     longest_seq_length, longest_seq_ix = get_longest_seq_length(train_data)
-    model.max_seq_length = longest_seq_length
+    model.max_seq_length = (
+        longest_seq_length if max_seq_length is None
+        else min(longest_seq_length, max_seq_length)
+    )
```

Comment:
Maybe this:
```suggestion
    model.max_seq_length = min(longest_seq_length, max_seq_length or float("inf"))
```
or it's too complicated?

Reply:
It looks reasonable to me, I'll change it